src/main/scala/chiseltest/simulator/jni/JniAPI.scala

```diff
@@ -0,0 +1,76 @@
+package chiseltest.simulator.jni
+
+import scala.io.Source
+
+object JniAPI {
+  /* Borrowed from JNAUtils */
+  val isWindows: Boolean = System.getProperty("os.name").toLowerCase().contains("win")
+
+  def javaHome: String = System.getProperty("java.home") match {
+    case s: String if s.endsWith("/jre") => s.dropRight(4)
+    case s: String                       => s
+  }
+
+  def osIncludeName: String = if (isWindows) {
+    "win32"
+  } else {
+    System.getProperty("os.name") match {
+      case "Mac OS X" => "darwin"
+      case "Linux"    => "linux"
+      case s: String => s
+    }
+  }
+
+  val tmpDir = os.Path(System.getProperty("java.io.tmpdir"))
+  val jniSoDir = tmpDir / "chiseltest_jni_bridge"
```

Comment:
Ideally you would compile everything in the local test directory instead of a temporary directory. That will make it easier for debugging.